File: services/api/src/models/unified_property.py
from typing import List, Optional, Dict, Any, Union
from pydantic import BaseModel, Field

class Coordinates(BaseModel):
    lat: float | None
    lng: float | None

class PricingModel(BaseModel):
    total: float

class CapacityModel(BaseModel):
    bedrooms: int
    beds: int

class FeaturesModel(BaseModel):
    size: Optional[float] = None
    amenities: List[str]

class MediaModel(BaseModel):
    main_image: str
    gallery: List[str]

class UnifiedProperty(BaseModel):
    # Basic Information
    id: str
    source: str  # 'Booking.com' or 'Airbnb'
    url: str

    # Property Details
    name: str
    description: str

    # Location Information
    location: str
    coordinates: Coordinates

    # Pricing Information
    pricing: PricingModel

    # Capacity and Layout
    capacity: CapacityModel

    # Property Features
    features: FeaturesModel

    # Media
    media: MediaModel

    # AI Scoring
    score: Union[int, float] = Field(ge=0, description="Score must be a non-negative number")
    reasoning: str

    # Raw Data
    raw_data: Optional[Any] = None

    class Config:
        allow_population_by_field_name = True

File: web/unified-property.ts
// Unified Property Structure for Front-end
export interface UnifiedProperty {
  // Basic Information
  id: string;                    // Unique identifier (could be from Booking or Airbnb)
  source: 'Booking.com' | 'Airbnb';  // Source platform
  url: string;                   // Original listing URL
  
  // Property Details
  name: string;                  // Property name/title
  description: string;           // Full description
  
  // Location Information
  location: string;

  // Pricing Information
  pricing: {
    total: number;
  };
  
  // Capacity and Layout
  capacity: {
    bedrooms: number;
    beds: number;
  };
  
  // Property Features
  features: {
    size?: number; // In square meters
    amenities: string[];
  };
  
  // Media
  media: {
    main_image: string;          // Changed from mainImage to match backend
    gallery: string[];           // Array of additional image URLs
  };

  score: number;               // AI-generated score (changed from string to number)
  reasoning: string;           // AI reasoning for the score
  
  // Raw Data (for debugging or advanced use)
  raw_data?: any;              // Changed from rawData to match backend
}